dev/release/download_rc_binaries.py

```diff
@@ -190,12 +192,18 @@ def _download_file(self, dest, asset):
             print("Already downloaded", dest_path)
             return
 
-        return self._download_url(
+        delay = random.randint(0, 3)
+        print(f"Waiting {delay} seconds to avoid rate limit")
+        time.sleep(delay)
+
+        self._download_url(
             url,
             dest_path,
             extra_args=[
                 "--header",
                 "Accept: application/octet-stream",
+                # Also retry 403s
+                "--retry-all-errors",
```

Comment:
It seems that we need to specify `--retry N` too:

https://curl.se/docs/manpage.html#--retry-all-errors

> Retry on any error. This option is used together with [--retry](https://curl.se/docs/manpage.html#--retry).

Reply:
Good catch!

Reply:
Oh wait, the `_download_url` implementation always passes `--retry 5`

https://github.com/apache/arrow/blob/8e9165e83eac7acd0137ef40f397dc05cfacd872/dev/release/download_rc_binaries.py#L115-L116